### PR TITLE
#244 conditional video processing

### LINKED ISSUE #244: Make a demo with a pyfunc setting a tag for drawing/encoding based on detected objects
The idea is simple. To give a user a universal function to add a tag when certain object classes are detected. 

The function must accept parameters in the form of `kwargs`:

```yaml
detections:
  - model: name1
    labels:
      - l1
      - l2
set_tags: 
  - tagname1
protection_interval_ms: 1000 # min 1 second switch period from state to state
```

Works on a principle: "set if any model/label pair matches".

Create a demo showing how peoplenet model with present and absent people triggers encoding and drawing (loop, module, always-on rtsp)

### PR DESCRIPTION
**Pull request by tomskikh**

### CODE REVIEW

File: samples/conditional_video_processing/demo.yml [hunk 1]
```diff
@@ -0,0 +1,179 @@
...
+      # thickness of counter's font
+      counters_font_thickness: 5
+    rendered_objects:
+      # Detector's element name from pipeline definition below
+      peoplenet:
+        # Detector's output object's label from pipeline definition
+        # Note that in this sample the analytics pyfunc assigns draw labels to objects
+        # If assigned, draw labels are used in place of the original object labels
+        # in the default draw func implementation
+        person_face:
```
Comment (bwsw) [inline samples/conditional_video_processing/demo.yml]: must be for 'person', not 'person_face'

File: samples/conditional_video_processing/demo.yml [hunk 2]
```diff
@@ -0,0 +1,179 @@
...
+          bbox:
+            # RGBA color value for bboxes of persons with face associated
+            border_color: '00ff00ff'
+            # width of person's bboxes border in pixels
+            thickness: 3
+          label:
+            # RGBA color values for person label's font and background
+            font_color: '000000ff'
+            background_color: 'd8e6ffd9'
+        person_noface:
```
Comment (bwsw) [inline samples/conditional_video_processing/demo.yml]: remove this.

File: samples/conditional_video_processing/demo.yml [hunk 3]
```diff
@@ -0,0 +1,179 @@
...
+                  min_height: 32
+            - class_id: 2
+              label: face
+              selector:
+                kwargs:
+                  # minimal detector confidence of the objects of this class to be included in output
+                  confidence_threshold: 0.1
+
+    # nvtracker element from Deepstream, provides tracking
+    - element: nvtracker
```
Comment (bwsw) [inline samples/conditional_video_processing/demo.yml]: remove it.

File: samples/conditional_video_processing/demo.yml [hunk 4]
```diff
@@ -0,0 +1,179 @@
...
+      properties:
+        tracker-width: 640
+        tracker-height: 384
+        ll-lib-file: /opt/nvidia/deepstream/deepstream/lib/libnvds_nvmultiobjecttracker.so
+        # specify tracking config file
+        ll-config-file: ${oc.env:PROJECT_PATH}/samples/peoplenet_detector/config_tracker_NvSORT.yml
+        enable_batch_process: 1
+
+    # analytics element realized in custom pyfunc
+    - element: pyfunc
```
Comment (bwsw) [inline samples/conditional_video_processing/demo.yml]: remove it

File: samples/conditional_video_processing/demo.yml [hunk 5]
```diff
@@ -0,0 +1,179 @@
...
+      top: 180
+      bottom: 0
+  output_frame:
+    codec: h264
+    # Encode only frames with tag "encode"
+    when_tagged: encode
+  # PyFunc for drawing on frames
+  draw_func:
+    # specify the drawfunc's python module
+    module: samples.peoplenet_detector.overlay
```
Comment (bwsw) [inline samples/conditional_video_processing/demo.yml]: remove custom drawfunc, default draw func must be used configured to draw only 'peoplenet.person' objects.

**Review by bwsw — CHANGES_REQUESTED** [review]

**Review by abramov-oleg — CHANGES_REQUESTED** [review]
Let's also add a README that briefly describes what the sample shows; mention that it is expected that the AOsink is regularly switching between video stream and stub img.

Add a link to the sample to the `samples/README.md`. Use the "Utility And Coding Examples" section.

**Review by bwsw — APPROVED** [review]

**Pull request merged**